Convert scripts in bin to console_scripts

rodrigc opened the pull request:
See:
http://twistedmatrix.com/trac/ticket/8491

rodrigc committed:
Use console_scripts entry_point for installing command-line scripts.

Remove getScripts()

rodrigc committed:
Remove tests for twisted.python.dist.getScripts()

twisted.python.dist.getScripts() was removed during
the conversion to the console_scripts entry point.

alex commented:
looks like some of the builders need to be updated

rodrigc committed:
Exclude admin and bin directories from the source checkout

rodrigc committed:
Do not overlap lists of scripts

rodrigc commented:
@adiroiban: why is codecov showing 71.42% coverage here: https://codecov.io/gh/twisted/twisted/compare/8dd2292cac9be85a9f81aef336c35bdb89f46e1c...87cb6c0d9af66591d218bd53684d58e15601daf4

It looks like it is not reporting coverage of the _PY3 block.  Isn't the https://buildbot.twistedmatrix.com/builders/fedora22-py3.4-coverage build uploading coverage results to codecov.io ?

alex commented on setup.py:
```diff
@@ -43,7 +43,9 @@ def main(args):
         conditionalExtensions=getExtensions(),
-        scripts=getScripts(),
+        entry_points={
+               'console_scripts':  getConsoleScripts()
```
This is the wrong indentation.

alex commented on setup3.py:
```diff
@@ -34,37 +34,26 @@ def find_package_modules(self, package, package_dir):
         conditionalExtensions=getExtensions(),
-        scripts=getScripts(),
+        entry_points={
+               'console_scripts':  getConsoleScripts()
```
Same here.

rodrigc committed:
Remove scripts from bin (except for trial)

These scripts are now installed by the console_scripts
entry point in setup.py.

rodrigc committed:
Add topfile
